x-pack/solutions/security/packages/kbn-scout-security/src/playwright/fixtures/test/page_objects/alerts_table.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ScoutPage, Locator } from '@kbn/scout';
+
+const PAGE_URL = 'security/alerts';
+
+export class AlertsTablePage {
+  public expandAlertBtn: Locator;
+
+  constructor(private readonly page: ScoutPage) {
+    this.expandAlertBtn = this.page.testSubj.locator('expand-event');
+  }
+
+  async navigate() {
+    await this.page.gotoApp(PAGE_URL);
+  }
+
+  async expandAlertDetailsFlyout(index = 0) {
+    await this.expandAlertBtn.nth(index).click();
+  }
```

Comment:
I believe using `nth` will be blocked by Playwright ESLint rules. Hope there is a way to make it work more robust?

Reply:
another learning from my side: Playwright won't try to validate if element is visible but will try to click it and may stuck for timeout.

Just FYI, I have do smth like 

```
    if (await this.toggleNavButton.isVisible()) {
      const isExpanded = await this.toggleNavButton.getAttribute('aria-expanded');
      if (isExpanded === 'false') {
        await this.toggleNavButton.click();
      }
    }
```

you need to simulate page not having button to see if it is needed